packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -1475,21 +1200,354 @@ class _DialState extends State<_Dial> with SingleTickerProviderStateMixin {
+                  const SizedBox(height: 8.0),
+                  if (!hourHasError && !minuteHasError)
+                    ExcludeSemantics(
+                      // TODO(rami-a): localize 'Minute'
+                      child: Text('Minute', style: theme.textTheme.caption, maxLines: 1, overflow: TextOverflow.ellipsis),
+                    ),
+                ],
+              )),
+              if (!use24HourDials && timeOfDayFormat != TimeOfDayFormat.a_space_h_colon_mm) ...<Widget>[
+                const SizedBox(width: 12.0),
+                _DayPeriodControl(
+                  selectedTime: _selectedTime,
+                  orientation: Orientation.portrait,
+                  onChanged: _handleDayPeriodChanged,
+                ),
+              ],
+            ],
+          ),
+          if (hourHasError || minuteHasError)
+            Text(
+              // TODO(rami-a): localize 'Enter a valid time'
+              'Enter a valid time',
+              style: theme.textTheme.bodyText2.copyWith(color: theme.colorScheme.error),
+            )
+          else
+            const SizedBox(height: 2.0),
+        ],
+      ),
+    );
+  }
+}
+
+class _HourMinuteTextField extends StatefulWidget {
+  const _HourMinuteTextField({
+    Key key,
+    @required this.selectedTime,
+    @required this.isHour,
+    @required this.style,
+    @required this.validator,
+    @required this.onSavedSubmitted,
+    this.onChanged,
+  }) : super(key: key);
+
+  final TimeOfDay selectedTime;
+  final bool isHour;
+  final TextStyle style;
+  final FormFieldValidator<String> validator;
+  final ValueChanged<String> onSavedSubmitted;
+  final ValueChanged<String> onChanged;
+
+  @override
+  _HourMinuteTextFieldState createState() => _HourMinuteTextFieldState();
+}
+
+class _HourMinuteTextFieldState extends State<_HourMinuteTextField> {
+  TextEditingController controller;
+  FocusNode focusNode;
+
+  @override
+  void initState() {
+    super.initState();
+    focusNode = FocusNode()..addListener(() {
+      setState(() { }); // Rebuild.
+    });
+  }
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    controller ??= TextEditingController(text: _formattedValue);
+  }
+
+  String get _formattedValue {
+    final bool alwaysUse24HourFormat = MediaQuery.of(context).alwaysUse24HourFormat;
+    final MaterialLocalizations localizations = MaterialLocalizations.of(context);
+    return widget.isHour ? localizations.formatHour(
+      widget.selectedTime,
+      alwaysUse24HourFormat: alwaysUse24HourFormat,
+    ) : localizations.formatMinute(widget.selectedTime);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final ThemeData theme = Theme.of(context);
+    final ColorScheme colorScheme = theme.colorScheme;
+
+    final InputDecorationTheme inputDecorationTheme = TimePickerTheme.of(context).inputDecorationTheme;
+    InputDecoration inputDecoration;
+    if (inputDecorationTheme == null) {
```

Comment:
This is OK but it makes it clear that the TimePickerTheme's inputDecorationTheme is underpowered - it can't reflect the time picker's state. Maybe make this a MaterialStateProperty instead.

Also: maybe InputDecorationTheme is more flexibility than is really needed here (I realize that it's convenient).  Maybe all that is needed are two properties: a MaterialStateProperty<BorderSide> and a MaterialStateProperty<Color>.  InputDecorationTheme implies more flexibility than we're likely to test or anyone is likely to need.

Reply:
Spoke offline and agreed that InputDecorationTheme provides enough flexibility to customize the TextFields if necessary